File: 125.valid-palindrome.java
/*
 * @lc app=leetcode id=125 lang=java
 *
 * [125] Valid Palindrome
 */

// @lc code=start
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase().replaceAll(" ", "").replaceAll("\\p{Punct}", "");
        for(int i = 0; i<(s.length()+1)/2; i++){
            if(s.charAt(i)!=s.charAt(s.length()-i-1)){
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

File: 125.valid-palindrome.py
import re
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        str = re.sub(r'[^A-Za-z0-9]', '', s).lower()
        for i in range(len(str)//2):
            if str[i] != str[-i-1]:
                return False
        return True
